vscode/src/azure/commands.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as vscode from "vscode";
+import { log } from "qsharp-lang";
+
+import {
+  Job,
+  Target,
+  WorkspaceTreeItem,
+  WorkspaceTreeProvider,
+} from "./treeView";
+import {
+  getJobFiles,
+  getTokenForWorkspace,
+  queryWorkspace,
+  queryWorkspaces,
+  submitJob,
+} from "./workspaceActions";
+import { QuantumUris, compileToBitcode } from "./networkRequests";
+import { getQirForActiveWindow } from "../qirGeneration";
+
+// TODO: Proper docs on the qsharp wiki
+const corsDocsUri =
+  "https://gist.github.com/billti/09637269db4bae86c0e3a552dd20eb9b";
+
+export function initAzureWorkspaces() {
+  const workspaceTreeProvider = new WorkspaceTreeProvider();
+  vscode.window.createTreeView("quantum-workspaces", {
+    treeDataProvider: workspaceTreeProvider,
+  });
+
+  vscode.commands.registerCommand(
+    "qsharp-vscode.targetSubmit",
+    async (arg: WorkspaceTreeItem) => {
+      const target = arg.itemData as Target;
+      const providerId = target.id.split(".")?.[0];
+
+      let qir = "";
+      try {
+        qir = await getQirForActiveWindow();
+      } catch (e: any) {
+        if (e?.name === "QirGenerationError") {
+          vscode.window.showErrorMessage(e.message);
+          return;
+        }
+      }
+      if (!qir) return;
+
+      // Note: Below compilation to be removed when all regions support .ll text directly
+      const compilerService: string | undefined = vscode.workspace
+        .getConfiguration("Q#")
+        .get("compilerService"); // e.g. in settings.json: "Q#.compilerService": "https://qsx-proxy.azurewebsites.net/api/compile"
```

Comment:
Was there supposed to be a separate consent setting for this service or is it covered by the `AzureQuantumIntegration` setting?

Reply:
This is now an 'undocumented' setting. It will also be deleted soon also once the latest service changes finish rolling out to production.